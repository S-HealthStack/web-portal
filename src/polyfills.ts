import smoothScroll from 'smoothscroll-polyfill';
import _isNull from 'lodash/isNull';
import _isObject from 'lodash/isObject';
import browser from 'src/common/utils/browser';

type ArgumentTypes<F extends () => unknown> = F extends (...args: infer A) => unknown ? A : never;

// safari smooth scroll fix
if (browser.isSafari) {
  smoothScroll.polyfill();

  const prevScrollTo = window.scrollTo;

  // issue: https://github.com/iamdustan/smoothscroll/issues/161
  Object.assign(window, {
    scrollTo: ((...args) => {
      if (_isObject(args[0])) {
        args[0].left = _isNull(args[0].left) ? undefined : args[0].left;
        args[0].top = _isNull(args[0].top) ? undefined : args[0].top;
      } else {
        args[0] = _isNull(args[0]) ? undefined : args[0];
        args[1] = _isNull(args[1]) ? undefined : args[1];
      }

      prevScrollTo(...(args as ArgumentTypes<typeof window.scrollTo>));
    }) as typeof window.scrollTo,
  });
}
