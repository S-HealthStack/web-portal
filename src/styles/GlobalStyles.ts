import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    font-size: 100%;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  html, body {
    height: 100%;
    width: 100%;
    margin: 0;
    position: fixed;
    overflow: hidden;
    overscroll-behavior-y: none;
    background: ${({ theme }) => theme.colors.background};
    user-select: none;
    -webkit-user-drag: none;
    font-family: Inter, sans-serif;
  }
  
  input,
  input:focus {
    ::-webkit-textfield-decoration-container {
      visibility: hidden;
    }
  }

  #root {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    overflow: hidden;
  }
  
  *:focus,
  *:focus-visible {
    outline: none;
  }
  
  .recharts-surface {
    overflow: visible;
  }

`;
