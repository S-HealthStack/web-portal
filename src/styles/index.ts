export { default as theme } from './theme';
export {
  colorsFn as colors,
  typographyFn as typography,
  boxShadowFn as boxShadow,
  animationFn as animation,
} from './themeHelpers';
export { default as GlobalStyles } from './GlobalStyles';
export { px } from './utils';
