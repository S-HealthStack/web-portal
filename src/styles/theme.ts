import { css, DefaultTheme } from 'styled-components';
import { px } from './utils';

const colors = {
  // gray gradations
  black: '#000000',
  black40: 'rgba(0, 0, 0, 0.4)',
  black08: 'rgba(0, 0, 0, 0.08)',

  // light mode
  surface: '#FFFFFF',
  background: '#F7F8FA',
  onPrimary: '#FFFFFF',
  backgroundOnPrimary: '#FFFFFF',
  backgroundSurface: '#FFFFFF',
  backgroundOnSurface: '#2D2D2D',
  primary: '#4475E3',
  primaryLight: '#ECF1FC',
  primaryHovered: '#2F65E0',
  primary05: '#F6F8FE',
  primary10: '#ECF1FC',
  primary30: '#C7D6F7',
  primaryDisabled: '#B3C6F1',
  primaryLightPressed: '#E3EAFB',
  primaryLightFocused: '#C7D6F7',
  textPrimary: '#474747',
  primaryWhite: '#FFFFFF',
  disabled: '#DBDBDB',
  disabled20: 'rgba(219, 219, 219, 0.2)',
  onDisabled: '#A1A1A1',

  textPrimaryBlue: '#4475E3',
  textPrimaryDark: '#474747',
  textSecondaryGray: '#808080',
  textDisabled: '#838383',
  onSurface: '#2D2D2D',
  secondarySkyblue: '#00B0D7', // replace
  backgoundLight: '#F0F0F0',
  primaryBlueHovered: '#2F65E0',
  primaryBluePressed: '#2057D5',

  statusSuccess: '#2DC008',
  statusSuccess10: '#DCF2EA',
  statusSuccessText: '#00825D',

  statusError: '#FF3333',
  statusError10: '#FFEBEB',
  statusErrorText: '#D14343',

  statusWarning: '#F5BF00',
  statusWarning10: '#FFF7D0',
  statusWarningText: '#A36400',

  secondaryViolet: '#944ED7',
  secondaryVioletHovered: '#6A3799',
  secondaryViolet10: '#F4EDFB',

  secondaryTangerine: '#F39C18',
  secondaryTangerineHovered: '#CC8314',
  secondaryTangerine10: '#FEF5E8',

  secondarySkyBlue: '#00B0D7',
  secondarySkyBlueHovered: '#0092B3',
  secondarySkyBlue10: '#E5F7FB',

  secondaryGreen: '#1AD598',
  secondaryGreenHovered: '#16B37F',
  secondaryGreen10: '#E8FBF5',

  secondaryRed: '#FA5F4F',
  secondaryRedHovered: '#CC4E40',
  secondaryRed10: '#FFEFED',
};

export type SpecColorType = keyof typeof colors;

const weightToCssValue = {
  regular: 400,
  upperregular: 450,
  medium: 500,
  semibold: 600,
  bold: 700,
};

const fontSpec = (
  size: number,
  lineHeight: number | string,
  weight: keyof typeof weightToCssValue,
  fontFamily: string
) => css`
  font-family: ${fontFamily}, sans-serif;
  font-size: ${px(size)};
  line-height: ${typeof lineHeight === 'string' ? lineHeight : px(lineHeight)};
  font-weight: ${weightToCssValue[weight]};
  font-style: normal;
`;

const typography = {
  // Inter typeface
  labelRegular: fontSpec(10, 15, 'regular', 'Inter'),
  labelSemibold: fontSpec(10, 15, 'semibold', 'Inter'),

  bodyXSmallRegular: fontSpec(12, 18, 'regular', 'Inter'),
  bodyXSmallSemibold: fontSpec(12, 18, 'semibold', 'Inter'),
  bodySmallRegular: fontSpec(14, 21, 'regular', 'Inter'),
  bodySmallSemibold: fontSpec(14, 21, 'semibold', 'Inter'),
  bodyMediumRegular: fontSpec(16, 24, 'regular', 'Inter'),
  bodyMediumSemibold: fontSpec(16, 24, 'semibold', 'Inter'),
  bodyLargeRegular: fontSpec(18, 26, 'regular', 'Inter'),
  bodyLargeSemibold: fontSpec(18, 26, 'semibold', 'Inter'),

  headingXSmall: fontSpec(16, 21, 'semibold', 'Inter'),
  headingSmall: fontSpec(18, 23, 'semibold', 'Inter'),
  headingXMedium: fontSpec(18, 27, 'semibold', 'Inter'),
  headingXMediumRegular: fontSpec(18, 27, 'regular', 'Inter'),
  headingMedium: fontSpec(24, 32.2, 'semibold', 'Inter'),
  headingLargeSemibold: fontSpec(32, 42, 'semibold', 'Inter'),
  headingXLargeSemibold: fontSpec(48, 58, 'semibold', 'Inter'),
  headingXXLargeSemibold: fontSpec(60, 72, 'semibold', 'Inter'),

  // Fira Code typeface
  query14: fontSpec(14, '130%', 'upperregular', 'Fira Code'),
  query12: fontSpec(12, '150%', 'upperregular', 'Fira Code'),
};

const boxShadow = {
  card: `0 0 ${px(2)} ${px(0)} rgba(0,0,0,0.15)`,
  avatar: `0 ${px(8)} ${px(10)} ${px(2)} 0 rgba(0,0,0,0.12)`,
  studyAvatar: `0 ${px(15)} ${px(20)} rgba(180,180,180,0.3)`,
  previewScreen: `0 ${px(2)} ${px(4)} rgba(71,71,71,0.25)`,
};

const animation = {
  defaultTiming: 'cubic-bezier(0.45, 0.05, 0, 1)',
  rippleTiming: 'cubic-bezier(0.4, 0, 0.2, 1)',
};

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof colors;
    typography: typeof typography;
    boxShadow: typeof boxShadow;
    animation: typeof animation;
  }
}

const theme: DefaultTheme = {
  colors,
  typography,
  boxShadow,
  animation,
};

export default theme;
