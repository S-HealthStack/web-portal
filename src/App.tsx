import React from 'react';
import { Provider } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'styled-components';
import { DndProvider } from 'react-dnd';
import { TouchBackend } from 'react-dnd-touch-backend';

import { store } from 'src/modules/store/store';
import { history, Path } from 'src/modules/navigation/store';
import PrivateRoute from 'src/modules/navigation/private-route/PrivateRoute';
import MainLayout from 'src/modules/main-layout/MainLayout';
import SignInScreen from 'src/modules/auth/signin/SignInScreen';
import AccountActivationScreen from 'src/modules/auth/signin/AccountActivationScreen';
import SignUp from 'src/modules/auth/signup/SignUp';
import CheckMailbox from 'src/modules/auth/signup/CheckMailbox';
import AccountCreated from 'src/modules/auth/signup/AccountCreated';
import CreateStudyScreen from 'src/modules/studies/CreateStudyScreen';
import { TooltipProvider, TooltipsList } from 'src/common/components/Tooltip';
import { theme, GlobalStyles } from 'src/styles';

import 'src/modules/auth/authProvider';

const App = () => (
  <ThemeProvider theme={theme}>
    <TooltipProvider>
      <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
        <Provider store={store}>
          <GlobalStyles />
          <ConnectedRouter history={history}>
            <Switch>
              <Route path={Path.SignIn} component={SignInScreen} />
              <Route path={Path.AccountActivation} component={AccountActivationScreen} />
              <Route path={Path.AccountCreate} component={SignUp} />
              <Route path={Path.AccountConfirm} component={CheckMailbox} />
              <Route path={Path.AccountVerification} component={AccountCreated} />
              <PrivateRoute path={Path.CreateStudy} component={CreateStudyScreen} />
              <PrivateRoute>
                <MainLayout />
              </PrivateRoute>
            </Switch>
          </ConnectedRouter>
        </Provider>
        <TooltipsList />
      </DndProvider>
    </TooltipProvider>
  </ThemeProvider>
);

export default App;
