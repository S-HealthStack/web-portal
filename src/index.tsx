import React from 'react';
import { createRoot } from 'react-dom/client';

import printAppVersion from 'src/common/utils/printAppVersion';

import './polyfills';
import App from './App';

printAppVersion();

const container = document.getElementById('root');
const root = createRoot(container as HTMLElement);

root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
