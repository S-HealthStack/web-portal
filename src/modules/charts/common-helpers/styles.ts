import styled, { css } from 'styled-components';

import { colors, typography, px, animation } from 'src/styles';
import { LINE_CLASS_NAME } from '../common-components/Line';
import { AREA_CLASS_NAME } from '../common-components/Area';
import { DOT_CLASS_NAME } from '../common-components/Dot';

interface ContainerProps {
  width: number;
  height: number;
}

export const DOTS_CONTAINER_CLASS_NAME = 'dots';
export const X_AXIS_CLASS_NAME = 'xAxis';
export const Y_AXIS_CLASS_NAME = 'yAxis';
export const ZOOM_CLASS_NAME = 'zoom';

export const Container = styled.div.attrs<ContainerProps>(({ width, height }) => ({
  style: {
    width: px(width),
    height: px(height),
    position: 'relative',
  },
}))<ContainerProps>``;

export const TooltipContent = styled.div`
  display: flex;
  align-items: center;
`;

export const TooltipNameBlock = styled.div`
  display: flex;
  margin-right: ${px(8)};

  p {
    ${typography.labelSemibold};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipDataBlock = styled.div`
  display: flex;
  flex-direction: column;

  p {
    ${typography.labelRegular};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipColorPoint = styled.div<{ color: string }>`
  width: ${px(12)};
  height: ${px(12)};
  background-color: ${({ color }) => color};
  border-radius: 50%;
  margin-right: ${px(8)};
`;

export const brushOverlayStyles = css`
  fill: ${colors.primary};
  fill-opacity: 0.1;
  pointer-events: none;
`;

export const brushSelectionStyles = css`
  fill: ${colors.primary};
  fill-opacity: 1;
`;

export const brushHandleCommonStyles = css`
  stroke-width: ${px(1)};
  stroke: ${colors.primary};
  stroke-opacity: 0.3;
  fill: ${colors.onPrimary};
`;

export const StyledSvg = styled.svg<{
  focusWidth: number;
  $contextVisible: boolean;
}>`
  .${X_AXIS_CLASS_NAME} {
    .tick line {
      opacity: 0;
    }

    .tick {
      :first-child text {
        text-anchor: start;
      }
      :last-child text {
        text-anchor: end;
      }
    }
  }

  .${Y_AXIS_CLASS_NAME} {
    .tick line {
      stroke: ${colors.updBackgoundLight};
    }
  }

  .context {
    opacity: ${({ $contextVisible }) => ($contextVisible ? 1 : 0)};
    transition: opacity 0.3s ${animation.defaultTiming};
    .brushX {
      .overlay {
        ${brushOverlayStyles};
        height: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        height: ${px(22)};
        fill: url(#horizontalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(22)};
        width: ${px(6)};
      }
    }
    .brushY {
      .overlay {
        ${brushOverlayStyles};
        width: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        width: ${px(22)};
        fill: url(#verticalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(6)};
        width: ${px(22)};
      }
    }
  }

  .focus {
    cursor: move;
  }

  .${LINE_CLASS_NAME} {
    clip-path: url(#clip);
    transition: stroke-opacity 0.3s ${animation.defaultTiming};
  }

  .${AREA_CLASS_NAME} {
    clip-path: url(#clip);
    transition: fill-opacity 0.3s ${animation.defaultTiming};
  }

  .${DOTS_CONTAINER_CLASS_NAME} {
    clip-path: url(#clip);
    .${DOT_CLASS_NAME} {
      transition: fill-opacity 0.3s ${animation.defaultTiming};
    }
  }

  .zoom {
    fill: none;
    pointer-events: all;
  }
`;
