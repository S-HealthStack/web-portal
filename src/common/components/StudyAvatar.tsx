import React from 'react';

import styled, { css, useTheme } from 'styled-components';

import StudyAvatarS from 'src/assets/icons/study_avatar_s.svg';
import StudyAvatarM from 'src/assets/icons/study_avatar_m.svg';
import StudyAvatarXXL from 'src/assets/icons/study_avatar_xxl.svg';
import SelectedStudyAvatarM from 'src/assets/icons/study_avatar_selected_m.svg';
import SelectedStudyAvatarXXL from 'src/assets/icons/study_avatar_selected_xxl.svg';
import { SpecColorType } from 'src/styles/theme';
import { animation, boxShadow, px } from 'src/styles';

export type AvatarSize = 's' | 'm' | 'xxl';

const adjustSizes = ($selected: boolean | undefined) => ({
  s: 36,
  m: $selected ? 69 : 62,
  xxl: $selected ? 147 : 142,
});

export type AvatarProps = {
  size?: AvatarSize;
  color: SpecColorType;
  faded?: boolean;
  $selected?: boolean;
} & React.HTMLAttributes<HTMLDivElement>;

const hoveredStyle = css`
  opacity: 1;
  box-shadow: ${boxShadow.studyAvatar};
  cursor: pointer;
`;

const pressedStyle = css`
  ${hoveredStyle};
  filter: drop-shadow(0 ${px(15)} ${px(20)} rgba(180, 180, 180, 0.3));
`;

const StyledAvatar = styled.div<AvatarProps>`
  border-radius: ${({ size }) => (size === 'xxl' ? px(66) : px(25))};
  height: ${({ size, $selected }) => size && px(adjustSizes($selected)[size])};
  transition: opacity 300ms ${animation.defaultTiming};
  position: relative;
  background-color: transparent !important;
  opacity: ${({ $selected }) => ($selected === false ? 0.5 : 1)};
  ${({ faded }) =>
    faded &&
    css`
      ${hoveredStyle}
    `};
  ${({ $selected, size }) =>
    $selected &&
    css`
      ${pressedStyle};
      height: ${px(adjustSizes($selected)[size || 'm'])};
    `};
  &:hover {
    ${hoveredStyle};
    &:active {
      ${pressedStyle};
    }
  }
`;

const AvatarIcon = (color: string, $selected: boolean | undefined) => ({
  s: <StudyAvatarS fill={color} data-testid="avatar-icon" />,
  m: $selected ? (
    <SelectedStudyAvatarM fill={color} data-testid="avatar-icon" />
  ) : (
    <StudyAvatarM fill={color} data-testid="avatar-icon" />
  ),
  xxl: $selected ? (
    <SelectedStudyAvatarXXL fill={color} data-testid="avatar-icon" />
  ) : (
    <StudyAvatarXXL fill={color} data-testid="avatar-icon" />
  ),
});

const StudyAvatar: React.FC<AvatarProps> = ({ color, size = 's', $selected, ...props }) => {
  const theme = useTheme();

  return (
    <StyledAvatar {...props} size={size} color={color} $selected={$selected}>
      {AvatarIcon(theme.colors[color], $selected)[size]}
    </StyledAvatar>
  );
};

export default StudyAvatar;
