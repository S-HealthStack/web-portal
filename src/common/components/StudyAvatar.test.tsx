import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { px } from 'src/styles';
import theme from 'src/styles/theme';
import { ThemeProvider } from 'styled-components/';
import StudyAvatar from './StudyAvatar';

describe('StudyAvatar', () => {
  it('test studyAvatar render', () => {
    const { baseElement, getByTestId, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <StudyAvatar color="primary" data-testid="styled-avatar" />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const avatar = getByTestId('styled-avatar');
    const avatarIcon = queryByTestId('avatar-icon') as Element;

    expect(avatar).toBeInTheDocument();
    expect(avatar).toHaveStyle(`height: ${px(40)}`);
    expect(avatarIcon.getAttribute('fill')).toBe(`${theme.colors.primary}`);
  });

  it('test studyAvatar selected state', () => {
    const { baseElement, getByTestId, rerender } = render(
      <ThemeProvider theme={theme}>
        <StudyAvatar color="primary" data-testid="styled-avatar" size="m" />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const avatar = getByTestId('styled-avatar');
    expect(avatar).toBeInTheDocument();
    expect(avatar).toHaveStyle(`height: ${px(62)}`);
    expect(avatar).not.toHaveStyle(`box-shadow: ${theme.boxShadow.studyAvatar}`);
    expect(avatar).not.toHaveStyle(
      `filter: drop-shadow(0 ${px(15)} ${px(20)} rgba(180,180,180,0.3))`
    );

    rerender(
      <ThemeProvider theme={theme}>
        <StudyAvatar color="primary" data-testid="styled-avatar" $selected size="m" />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    expect(avatar).toBeInTheDocument();
    expect(avatar).toHaveStyle(`height: ${px(69)}`);
    expect(avatar).toHaveStyle(`box-shadow: ${theme.boxShadow.studyAvatar}`);
    expect(avatar).toHaveStyle(`filter: drop-shadow(0 ${px(15)} ${px(20)} rgba(180,180,180,0.3))`);
  });

  it('[NEGATIVE] should render with wrong props', () => {
    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <StudyAvatar color={'unknown' as 'primary'} data-testid="styled-avatar" />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
    expect(getByTestId('styled-avatar')).toBeInTheDocument();
  });

  it('[NEGATIVE] should render without props', () => {
    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <StudyAvatar data-testid="styled-avatar" />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
    expect(getByTestId('styled-avatar')).toBeInTheDocument();
  });
});
