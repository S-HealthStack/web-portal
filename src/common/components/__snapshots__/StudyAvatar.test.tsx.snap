// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudyAvatar [NEGATIVE] should render with wrong props 1`] = `
.c0 {
  border-radius: 1.5625rem;
  height: 2.5rem;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  position: relative;
  background-color: transparent !important;
  opacity: 1;
}

.c0:hover {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
}

.c0:hover:active {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
}

<body>
  <div>
    <div
      class="c0"
      color="unknown"
      data-testid="styled-avatar"
    >
      <svg
        data-testid="avatar-icon"
      />
    </div>
  </div>
</body>
`;

exports[`StudyAvatar [NEGATIVE] should render without props 1`] = `
.c0 {
  border-radius: 1.5625rem;
  height: 2.5rem;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  position: relative;
  background-color: transparent !important;
  opacity: 1;
}

.c0:hover {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
}

.c0:hover:active {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
}

<body>
  <div>
    <div
      class="c0"
      data-testid="styled-avatar"
    >
      <svg
        data-testid="avatar-icon"
        fill="#000000"
      />
    </div>
  </div>
</body>
`;

exports[`StudyAvatar test studyAvatar render 1`] = `
.c0 {
  border-radius: 1.5625rem;
  height: 2.5rem;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  position: relative;
  background-color: transparent !important;
  opacity: 1;
}

.c0:hover {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
}

.c0:hover:active {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
}

<body>
  <div>
    <div
      class="c0"
      color="primary"
      data-testid="styled-avatar"
    >
      <svg
        data-testid="avatar-icon"
        fill="#4475E3"
      />
    </div>
  </div>
</body>
`;

exports[`StudyAvatar test studyAvatar selected state 1`] = `
.c0 {
  border-radius: 1.5625rem;
  height: 3.875rem;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  position: relative;
  background-color: transparent !important;
  opacity: 1;
}

.c0:hover {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
}

.c0:hover:active {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
}

<body>
  <div>
    <div
      class="c0"
      color="primary"
      data-testid="styled-avatar"
    >
      <svg
        data-testid="avatar-icon"
        fill="#4475E3"
      />
    </div>
  </div>
</body>
`;

exports[`StudyAvatar test studyAvatar selected state 2`] = `
.c0 {
  border-radius: 1.5625rem;
  height: 4.3125rem;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  position: relative;
  background-color: transparent !important;
  opacity: 1;
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  height: 4.3125rem;
}

.c0:hover {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
}

.c0:hover:active {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
}

<body>
  <div>
    <div
      class="c0"
      color="primary"
      data-testid="styled-avatar"
    >
      <svg
        data-testid="avatar-icon"
        fill="#4475E3"
      />
    </div>
  </div>
</body>
`;
